libgalois/include/katana/PropertyGraph.h

```diff
@@ -97,6 +98,34 @@ struct KATANA_EXPORT GraphTopology {
 /// manages the serialization of the various partitions and properties that
 /// comprise the physical representation of the logical property graph.
 class KATANA_EXPORT PropertyGraph {
+public:
+  /// TypeSetID uniquely identifies/contains a combination/set of types
+  /// TypeSetID is represented using 8 bits
+  /// TypeSetID for nodes is distinct from TypeSetID for edges
+  using TypeSetID = uint8_t;
+  static constexpr TypeSetID kUnknownType = TypeSetID{0};
+  static constexpr TypeSetID kInvalidType =
+      std::numeric_limits<TypeSetID>::max();
+  /// A set of TypeSetIDs
+  using SetOfTypeSetIDs =
+      std::bitset<std::numeric_limits<TypeSetID>::max() + 1>;
+  /// A set of type names
+  using SetOfTypeNames = std::unordered_set<std::string>;
+  /// A map from TypeSetID to the set of the type names it contains
+  using TypeSetIDToSetOfTypeNamesMap = std::vector<SetOfTypeNames>;
+  /// A map from the type name to the set of the TypeSetIDs that contain it
+  using TypeNameToSetOfTypeSetIDsMap =
+      std::unordered_map<std::string, SetOfTypeSetIDs>;
+
+  // Pass through topology API
+  using node_iterator = GraphTopology::node_iterator;
+  using edge_iterator = GraphTopology::edge_iterator;
+  using edges_range = GraphTopology::edges_range;
+  using iterator = GraphTopology::iterator;
+  using Node = GraphTopology::Node;
+  using Edge = GraphTopology::Edge;
+
```

Comment:
This is one of the few remaining places that is not using gluon's `PartitionTypes.h`.  Has the time come?

Reply:
This is in the open repo. So we cannot use anything from gluon. What should probably happen is that the types should be moved here and then referenced from gluon. This is more basic than gluon really.